File: C/Exercice2_4.c
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>

int main()
{
    __pid_t pid_fils;

    for(int i = 0; i < 5; i++)
    {
        pid_fils = fork();
        if(pid_fils == -1)
        {
            puts("Erreur lors de la creation du nouveau processus\n");
            exit(1);
        }
        if(pid_fils == 0)
        {
            printf("Je suis le fils %d et mon PID est: %d\n", i+1, getpid());
            exit(0);
        }
    }
    sleep(1);
    printf("Je suis le pere et mon PID est: %d\n", getpid());

    return 0;
}

File: Python/Exercice2_4.py
#coding: utf-8
import os
import sys
import time

for i in range(5):
    pid_fils = os.fork()
    
    if pid_fils == -1:
        print("Erreur lors de la creation du nouveau processus\n")
        sys.exit(1)

    if pid_fils == 0:
        print(f"Je suis le fils {i+1} et mon PID est: {os.getpid()}")
        sys.exit(0)

time.sleep(1)
print(f"Je suis le pere et mon PID est: {os.getpid()}")